### PR TITLE
PLATFORM-907: double check in logSql for bool value

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-907

@michalroszka / @owend

### CODE REVIEW

**Comment by michalroszka** [comment]
Good call updating the documentation, as well!

File: includes/db/Database.php [hunk 1]
```diff
@@ -3521,23 +3521,28 @@ protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
...
 		global $wgDBcluster;
 
-		if ( $ret === false ) {
-			return;
+		if ( $ret instanceof ResultWrapper ) {
+			$num_rows = $ret->numRows();
+		} elseif ( is_resource( $ret ) ) {
+			$num_rows = mysql_num_rows( $ret );
+		} else {
+			// INSERT, UPDATE, DELETE, DROP queries do not report num_rows
```
Comment (michalroszka) [inline includes/db/Database.php]: Although we can grab `mysql_affected_rows` from the connection resource, if available in this scope.

Comment (macbre) [inline includes/db/Database.php]: Good point, added

File: includes/db/Database.php [hunk 2]
```diff
@@ -3521,23 +3521,33 @@ protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
...
 		global $wgDBcluster;
 
-		if ( $ret === false ) {
-			return;
+		if ( $ret instanceof ResultWrapper ) {
+			// NOP for MySQL driver
+			$num_rows = $ret->numRows();
+		} elseif ( is_resource( $ret ) ) {
+			// for SELECT queries report how many rows are sent to the client
+			$num_rows = mysql_num_rows($ret);
```
Comment (michalroszka) [inline includes/db/Database.php]: MW coding style, whitespace :-)

File: includes/db/Database.php [hunk 3]
```diff
@@ -3500,14 +3500,14 @@ public function setBigSelects( $value = true ) {
...
 	protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
 		$queryId = MWDebug::query( $sql, $fname, $isMaster );
 		$start = microtime(true);
 
 		// Wikia change: DatabaseMysql returns a resource instead of ResultWrapper instance
-		/* @var $ret resource */
+		/* @var $ret resource|bool */
 		$ret = $this->doQuery( $sql );
 
 		$this->logSql( $sql, $ret, $fname, microtime(true) - $start, $isMaster );
```
Comment (michalroszka) [inline includes/db/Database.php]: MW coding style, whitespace :-)

File: includes/db/Database.php [hunk 4]
```diff
@@ -3500,14 +3500,14 @@ public function setBigSelects( $value = true ) {
 	 * @param string $fname the function name
 	 * @param bool $isMaster is this against the master
 	 *
-	 * @return ResultWrapper see doQuery
+	 * @return ResultWrapper|resource|bool see doQuery
 	 */
 	protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
 		$queryId = MWDebug::query( $sql, $fname, $isMaster );
 		$start = microtime(true);
```
Comment (michalroszka) [inline includes/db/Database.php]: MW coding style, whitespace :-)

**Comment by michalroszka** [comment]
Now, looks purrfect :)

**Comment by macbre** [comment]
PR well polished :)

**Pull request merged**